### DISCUSSION TITLE
Github.ref is not set when using workflow_dispatch

### DISCUSSION THREAD

**Discussion by matbour** [community/community#25958] [Actions]
<p>The workflow_dispatch feature was released a few days ago.<br>
To call the API endpoint, you have to provide a got ref ; however in the triggered workflow, the github.ref context is not set.<br>
My current workaround is to pass my ref into the workflow input, but it sounds like a hack to me.<br>
Has somebody experienced this issue?</p>

**Comment by Yanjingzhu** [comment]
<p>Thank you for pointing this out. I could reproduce your issue in my side. When create workflow_dispatch event through <a href="https://docs.github.com/en/rest/reference/actions#create-a-workflow-dispatch-event" rel="nofollow noopener">this</a> api, github.ref is empty.  I have reported it to GitHub engineering team, let’s wait for their response.</p>

**Comment by matbour** [comment]
<p>Thank you <a class="mention" href="/yanjingzhu">@yanjingzhu</a>. Do you have any link where I can follow this progress of this issue?</p>

**Comment by Yanjingzhu** [comment]
<p><a class="mention" href="/mathieu-bour">@mathieu-bour</a> Sorry about that I reported in GitHub internal channel. The link is not public. I will keep you posted when there are any updates in that internal ticket.</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p><a class="mention" href="/mathieu-bour">@mathieu-bour</a> Good news! This issue has been fixed.<br>
Please try again in your side and kindly let me know your result.</p>

**Reply by jpmhouston** [reply]
What is github.ref set to when workflow_dispatch is the trigger? Where in the documentation can we find information about this.

More information would have been helpful to post here rather than just the minimal message "it's fixed". Please consider people searching for these topics after the fact.
